### PR TITLE
fix(kit): don't allow disabled options to be selected

### LINKED ISSUE #5965: 🐞 - It's possible to select disabled option in a Combobox
### Playground Link

https://angular-tui-combobox-issue.stackblitz.io

### Description

Use the StackBlitz link provided above. Type the exact string to match the disabled name from the list into the input field. When you focus out and then focus in again, the disabled value will be selected.

### Angular version

15.2.9

### Taiga UI version

3.28.0

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [X] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

Closes #5965

### CODE REVIEW

File: projects/core/components/data-list/option/option.component.ts [hunk 1]
```diff
@@ -86,7 +86,7 @@ export class TuiOptionComponent<T = unknown> implements OnDestroy {
 
     @HostListener('click')
     onClick(): void {
-        if (this.host && this.value !== undefined) {
+        if (this.host && this.value !== undefined && !this.disabled) {
```
Comment (waterplea) [inline projects/core/components/data-list/option/option.component.ts:89]: I don't think this is necessary, it is covered by pointer events and native disable of the button.

File: projects/demo/src/modules/components/combo-box/combo-box.template.html [hunk 2]
```diff
@@ -89,6 +89,14 @@
         >
             <tui-combo-box-example-6></tui-combo-box-example-6>
         </tui-doc-example>
+
+        <tui-doc-example
+            id="ignore-disabled"
+            heading="Don't allow disabled options to be selected"
```
Comment (waterplea) [inline projects/demo/src/modules/components/combo-box/combo-box.template.html:95]: Let's add `by typing them`

File: projects/demo/src/modules/components/combo-box/examples/7/index.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<form class="b-form">
+    <tui-combo-box
+        class="b-form"
+        [ngModelOptions]="{standalone: true}"
```
Comment (waterplea) [inline projects/demo/src/modules/components/combo-box/examples/7/index.html:4]: You can just remove wrapping `form` tag, it's unnecessary here

File: projects/demo/src/modules/components/combo-box/examples/7/index.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        />
+        <ng-container *tuiDataList>
+            <tui-data-list>
+                <button
+                    *ngFor="let item of items | tuiFilterByInputWith: stringify"
+                    tuiOption
+                    [disabled]="item.disabled"
+                    [value]="item"
+                >
+                    {{ item | tuiMapper: stringify }}
```
Comment (waterplea) [inline projects/demo/src/modules/components/combo-box/examples/7/index.html:21]: Piping is unnecessary for such simple operation.

**Review by waterplea — APPROVED** [review]

File: projects/demo/src/modules/components/combo-box/examples/7/index.html (hunk 4 above)
Comment (splincode) [reply projects/demo/src/modules/components/combo-box/examples/7/index.html]: fixed

File: projects/demo/src/modules/components/combo-box/examples/7/index.html (hunk 3 above)
Comment (splincode) [reply projects/demo/src/modules/components/combo-box/examples/7/index.html]: fixed

File: projects/demo/src/modules/components/combo-box/combo-box.template.html (hunk 2 above)
Comment (splincode) [reply projects/demo/src/modules/components/combo-box/combo-box.template.html]: fixed

File: projects/core/components/data-list/option/option.component.ts (hunk 1 above)
Comment (splincode) [reply projects/core/components/data-list/option/option.component.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**